src/test/scala/com/microsoft/hyperspace/index/IndexLogEntryTest.scala

```diff
@@ -139,12 +178,203 @@ class IndexLogEntryTest extends SparkFunSuite {
+    val content = Content.fromDirectory(dir2Path)
+
+    val expected = Seq(f3, f4).map(toPath).toSet
+    val actual = content.files.toSet
+    assert(actual.equals(expected))
+  }
+
+  test("Content Test: files api lists all files recursively from Content object.") {
+    val dirPath = PathUtils.makeAbsolute(testDir.toString)
+    val content = Content.fromDirectory(dirPath)
+
+    val expected = Seq(f1, f2, f3, f4).map(toPath).toSet
+    val actual = content.files.toSet
+    assert(actual.equals(expected))
+  }
+
+  test("Content Test: fromDirectory api creates the correct Content object.") {
+    val dir2Path = PathUtils.makeAbsolute(nestedDir.toString)
+
+    val fileInfos = Seq(f3, f4).map(toFileStatus).map(FileInfo(_))
+
+    val expected = {
+      val leafDir = Directory("nested", fileInfos)
+      val rootDirectory = TestUtils.splitPath(dir2Path.getParent).foldLeft(leafDir) {
+        (accum, name) =>
+          Directory(name, Seq(), Seq(accum))
+      }
+      Content(rootDirectory, NoOpFingerprint())
+    }
+
+    val actual = Content.fromDirectory(dir2Path)
+    assert(actual.equals(expected))
+  }
+
+  test("Content Test: fromLeafFiles api creates the correct Content object.") {
+    val dir2Path = PathUtils.makeAbsolute(nestedDir.toString)
+    val fileStatuses = Seq(f3, f4).map(toFileStatus)
+
+    val expected = {
+      val fileInfos = fileStatuses.map(FileInfo(_))
+      val leafDir = Directory("nested", fileInfos)
+      val rootDirectory = TestUtils.splitPath(dir2Path.getParent).foldLeft(leafDir) {
+        (accum, name) =>
+          Directory(name, Seq(), Seq(accum))
+      }
+      Content(rootDirectory, NoOpFingerprint())
+    }
+
+    val actual = Content.fromDirectory(dir2Path)
+    assert(actual.equals(expected))
+  }
+
+  test("Directory Test: fromDirectory api creates the correct Directory object.") {
+    val dir2Path = PathUtils.makeAbsolute(nestedDir.toString)
+
+    val fileInfos = Seq(f3, f4).map(toFileStatus).map(FileInfo(_))
+
+    val expected = {
+      val leafDir = Directory("nested", fileInfos)
+      TestUtils.splitPath(dir2Path.getParent).foldLeft(leafDir) {
+        (accum, name) =>
+          Directory(name, Seq(), Seq(accum))
+      }
+    }
+
+    val actual = Directory.fromDirectory(dir2Path)
+    assert(actual.equals(expected))
+  }
+
+  test("Directory Test: fromDirectory api creates the correct Directory objects, " +
+    "recursively listing all leaf files.") {
+    val dirPath = PathUtils.makeAbsolute(testDir.toString)
+    val nestedDirPath = PathUtils.makeAbsolute(nestedDir.toString)
+
+    val testDirLeafFiles = Seq(f1, f2).map(toFileStatus).map(FileInfo(_))
+    val nestedLeafFiles = Seq(f3, f4).map(toFileStatus).map(FileInfo(_))
+    val leafDir1 = Directory(name = "testDir", files = testDirLeafFiles)
+    val leafDir2 = Directory(name = "nested", files = nestedLeafFiles)
+    val expected2 = TestUtils.splitPath(dirPath.getParent).foldLeft(leafDir1) { (accum, name) =>
```

Comment:
Can you refactor `splitPath... foldLeft` to a function with a sensible name instead of repeating these?

Reply:
thanks @imback82 , done